src/lints/trait_removed_associated_type.ron

```diff
@@ -0,0 +1,54 @@
+SemverQuery(
+    id: "trait_removed_associated_type",
+    human_readable_name: "associated type of trait removed",
+    description: "A trait's associated type was removed.",
+    required_update: Major,
+    reference_link: None,
```

Comment:
We only really want to leave this as `None` if there's literally no link that mentions anything whatsoever that would be useful to know if this lint fires.

Fortunately, there is a useful thing: associated types are items, and the semver reference has an entry about removing items being a breaking change that we've used in multiple other lints.

Reply:
[05dcb37](https://github.com/obi1kenobi/cargo-semver-checks/pull/500/commits/05dcb3719dcadac5189160637060195b77ac40b3)